scalding-core/src/main/scala/com/twitter/scalding/typed/KeyedList.scala

```diff
@@ -303,6 +303,19 @@ trait KeyedListLike[K, +T, +This[K, +T] <: KeyedListLike[K, T, This]]
 
   /** For each key, give the number of values */
   def size: This[K, Long] = mapValues { x => 1L }.sum
+
+  /**
+   * For each key, give the number of unique values. WARNING: May OOM.
+   * This assumes the values for each key can fit in memory.
+   */
+  def distinctSize[B >: T]: This[K, Long] = toSet[B].mapValues(_.size)
```

Comment:
Yup, this should work. Just updated the code.

On Thu, Mar 26, 2015 at 10:08 AM, P. Oscar Boykin <notifications@github.com>
wrote:

> In scalding-core/src/main/scala/com/twitter/scalding/typed/KeyedList.scala
> <https://github.com/twitter/scalding/pull/1232#discussion_r27233370>:
>
> > @@ -303,6 +303,19 @@ trait KeyedListLike[K, +T, +This[K, +T] <: KeyedListLike[K, T, This]]
> >
> >    /** For each key, give the number of values */
> >    def size: This[K, Long] = mapValues { x => 1L }.sum
> > +
> > +  /**
> > +   * For each key, give the number of unique values. WARNING: May OOM.
> > +   * This assumes the values for each key can fit in memory.
> > +   */
> > +  def distinctSize[B >: T]: This[K, Long] = toSet[B].mapValues(_.size)
>
> def distinctSize: This[K, Long] = toSet[T].mapValues(_.size)
>
> should work, right?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/twitter/scalding/pull/1232/files#r27233370>.
>